lib/matplotlib/axes/_axes.py

```diff
@@ -6756,9 +6756,11 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
 
         if label is None:
             labels = [None]
-        elif isinstance(label, str):
-            labels = [label]
         else:
+            if isinstance(label, str):
```

Comment:
Rather keep it on one level:

~~~
        if label is None:
            labels = [None]
        elif isinstance(label, str):
            labels = [label]
        elif not np.iterable(label):
            labels = [str(label)]
        else:
            labels = [str(lab) for lab in label]
~~~

Reply:
Sure - I was hoping it'd have been even simpler, but `str(None)` returns `"None"`, which ruined my plans.